Src/coffeescript/cql-execution/src/elm/quantity.coffee

```diff
@@ -28,16 +28,31 @@ module.exports.Quantity = class Quantity extends Expression
     if other instanceof Quantity and other.unit == @unit then @value > parseFloat other.value else null
 
   before: (other) ->
-    if other instanceof Quantity and other.unit == @unit then @value < parseFloat other.value else null
+    if other instanceof Quantity and other.unit == @unit 
+      @value < parseFloat other.value 
+    else if other instanceof Quantity and other.unit of time_units and @unit of time_units
+      thisSmallestDuration = smallestDuration(@)
+      otherSmallestDuration = smallestDuration(other)
+      thisSmallestDuration < otherSmallestDuration
+    else null
     
   equals: (other) ->
     if other.unit? && other.value? then @unit == other.unit && @value == parseFloat other.value else null
 
-time_units = {'years':'year',  'months': 'month',  'days' :'day', 'minutes': 'minute', 'seconds':'seconds', 'milliseconds' : 'millisecond' }
+time_units = {'years': 'year', 'yr': 'year', 'y': 'year', 'months': 'month', 'weeks': 'week', 'wk': 'week', 'wks': 'week', 'days': 'day', 'd': 'day', 'minutes': 'minute', 'min': 'minute', 'seconds':'second', 'sec':'second', 'second':'second', 'milliseconds' : 'millisecond' }
```

Comment:
This is done to deal with the fact that at the moment the *units* field of a Result Quantity in Bonnie is a freetext field.  Ideally we would have a dropdown (or some other method) to limit the units that users can use in Bonnie.